### PR TITLE
Improve Benchmark

### PR DESCRIPTION
**Pull request by LucGenetier**
Improve benchmark project
Add reference tests
Add script to run locally easily
Add script to read results quickly
Add script to write results to SQL database

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Performance.Tests/Reference.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        public int FixedDuration()
+        {
+            // Fixed duration, independant of CPU speed 
+            Thread.Sleep(100);
+            return 0;
+        }
+
+        [Benchmark]
+        [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
+        public long Loop()
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Performance.Tests/Reference.cs:49]: Can we rename to "FixedLoop".

And add a comment in this class that the goal for all these tests is that they should take constant time (independent of any Power Fx changes), and so act as a baseline reference for performance.

File: src/Directory.Build.props [hunk 2]
```diff
@@ -73,7 +73,7 @@
     <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(Project)'!='Microsoft.PowerFx.Performance.Tests'">
+  <PropertyGroup Condition=" '$(ProjectName)' != 'Microsoft.PowerFx.Performance.Tests' ">
```
Comment (MikeStall) [inline src/Directory.Build.props:76]: Curious: what is project vs. projectname?

**Review by MikeStall — APPROVED** [review]

**Review by MikeStall — DISMISSED** [review]

File: src/Directory.Build.props (hunk 2 above)
Comment (LucGenetier) [reply src/Directory.Build.props:76]: ProjectName is documented
https://learn.microsoft.com/en-us/cpp/build/reference/common-macros-for-build-commands-and-properties?redirectedfrom=MSDN&view=msvc-170

File: src/tests/Microsoft.PowerFx.Performance.Tests/Reference.cs (hunk 1 above)
Comment (LucGenetier) [reply src/tests/Microsoft.PowerFx.Performance.Tests/Reference.cs]: Done

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**